cmd/rokka/cli/massupload/worker.go

```diff
@@ -0,0 +1,62 @@
+package massupload
+
+import (
+	"github.com/rokka-io/rokka-go/rokka"
+	"fmt"
+	"os"
+	"path/filepath"
+	"math/rand"
+)
+
+func UploadWorker(client *rokka.Client, imageDetails chan ImageDetails, results chan UploadResult, quit chan struct{}, dryRun bool) {
+	// TODO: handle worker names?
+	id := fmt.Sprintf("%.2d", rand.Uint32() % 100)
+
+	// fmt.Printf("Worker[%s]: Starting\n", id)
+	for imgDetail := range imageDetails {
+		// Actual upload the image
+		// fmt.Printf("Worker[%s]: Uploading %s\n", id, imgDetail)
+
+		result := UploadResult{
+			Path: imgDetail.Path,
+		}
+
+		if dryRun {
+			// fmt.Printf("Worker[%s]: DRY-RUN, upload skipped!\n", id)
+			result.RokkaHash = "DRY-RUN-HASH"
+		} else {
+			result.RokkaHash, result.Error = executeRokkaUpload(client, imgDetail, id)
+		}
+
+		select {
+		case results <- result:
+			// Send the result message
+		case <- quit:
+			// If we get a quit message, end the worker!
+			return
+		default:
+			// Do nothing, just fetch tne next image to upload!
+		}
+	}
+
+	// fmt.Printf("Worker[%s]: EXIT\n", id)
+}
+
+func executeRokkaUpload(client *rokka.Client, imgDetail ImageDetails, id string) (string, error) {
+	file, err := os.Open(imgDetail.Path)
+	if err != nil {
+		return "", err
+	}
+
+	// Postpone the file closing
+	defer file.Close()
+
+	// fmt.Printf("Worker[%s]: File opened, pushing to Rokka...\n", id)
+	response, err := client.CreateSourceImageWithMetadata(imgDetail.Organization, filepath.Base(imgDetail.Path), file, imgDetail.UserMetadata, nil)
+	if err != nil {
+		// fmt.Printf("Got error from Rokka upload: %v\n", err)
+		return "", err
+	}
+
+	return response.Items[0].Hash, nil
```

Comment:
I'd return everything. The template can then take care of only showing what's wanted.